Examples/Statistics/ImageHistogram1.cxx

```diff
@@ -84,17 +84,24 @@ main(int argc, char * argv[])
 
   // Software Guide : BeginLatex
   //
-  // Using the same image type we instantiate the type of the image reader
+  // Using the same image type we read the image file
   // that will provide the image source for our example.
+  // As usual, this must be done inside a try/catch block because the read
+  // operation can potentially throw exceptions.
   //
   // Software Guide : EndLatex
 
   // Software Guide : BeginCodeSnippet
-  using ReaderType = itk::ImageFileReader<ImageType>;
-
-  auto reader = ReaderType::New();
-
-  reader->SetFileName(argv[1]);
+  try
+  {
+    const auto input = itk::ReadImage<ImageType>(argv[1]);
```

Comment:
Here the input has very narrow scope. It is available only between lines 96 and 98.